togglebutton.js

```diff
@@ -0,0 +1,77 @@
+const template = document.createElement("template");
+template.innerHTML = `
+<style>
+.slider {
+    position: absolute;
+    width: 20px;
+    height: 20px;
+    background-color: #808080;
+    border-radius: 50%;
+    transition-duration: 1s;
+    pointer-events: none;
+  }
+  
+  .container {
+    width: 40px;
+    height: 20px;
+    background-color: black;
+    border-radius: 30% / 50%;
+    position: relative;
+    pointer-events: none;
+  }
+  
+  .slide {
+    transform: translateX(100%);
+  }
+  
+  .container input {
+    opacity: 0;
+    position: absolute;
+    pointer-events: all;
+  }
+  </style>
+            <div class="container">
+            <input  type="checkbox"  id="something"/>
+              <div
+                class="slider"
+                
+              ></div>
+            </div>
+`;
+class ToggleButton extends HTMLElement {
+  static get observedAttributes() {
+    return ["on"];
+  }
+  constructor() {
+    super();
+    this.attachShadow({ mode: "open" });
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+  }
+
+  connectedCallback() {
+    const radiobutton = this.shadowRoot.querySelector(".slider");
+    if (this.hasAttribute("on")) {
```

Comment:
Nice. Let's take this further and try adding a getter/setter to allow the following:

```tsx
let toggle = document.createElement("anveshmekala-toggle-button");
toggle.on = true; // note that we are setting the property here and not the attribute
document.body.append(toggle);
```

Reply:
get/set would enhance the  usability. Made the change  in [9b6960a](https://github.com/anveshmekala/native-components/pull/1#:~:text=feedback%20changes-,9b6960a,-Merge%20state)